src/debug.c

```diff
@@ -32,39 +32,120 @@
 /* ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE */
 /* POSSIBILITY OF SUCH DAMAGE. */
 
+#include <stdarg.h>
+#include <stdbool.h>
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 
 #include "debug.h"
 
-void dbg_print_matrix_d3(double const a[3][3]) {
-    int i;
-    for (i = 0; i < 3; i++) {
-        printf("%f %f %f\n", a[i][0], a[i][1], a[i][2]);
+bool debug_enabled() {
+    char const* debug_env = getenv("SPGLIB_DEBUG");
+    // If SPGLIB_DEBUG is not defined, do not output any debug info
+    if (debug_env == NULL) return false;
+    // Here we are not checking if SPGLIB_DEBUG is true/1/etc. we only check if
+    // it is defined, including SPGLIB_DEBUG=""
+    return true;
+}
+bool warning_enabled() {
+    char const* warning_env = getenv("SPGLIB_WARNING");
+    // If SPGLIB_WARNING is not defined assume warning is on
+    if (warning_env == NULL) return true;
+    // Check if SPGLIB_WARNING is disabled. Not performing case-insensitive
+    // checks.
+    if (strcmp(warning_env, "OFF") == 0) return false;
```

Comment:
Why only `warning_enabled` accepts "OFF" compared to `debug_enabled` and `info_enabled`?

Reply:
The others are `default: false`, while this is the only one to be `default: true`. For default false, we can assume if you set an environment variable the intent is to enable it. For the opposite case it is more ambiguous, that's why the explicit check of the value.